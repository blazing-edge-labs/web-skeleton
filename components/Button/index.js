import React from 'react';
import PropTypes from 'prop-types';

import './style.local.scss';

const Button = ({ children, ...rest }) =>
  <button
    styleName="button"
    {...rest}
  >
    {children}
  </button>;

Button.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.element,
  ]).isRequired,
};

export default Button;
