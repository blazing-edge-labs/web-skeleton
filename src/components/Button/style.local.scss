.button {
  & {
    height: 40px;
    width: 160px;
    background: #DBDBDB;
    border: 1px solid #000;
    border-radius: 2px;
  }

  &:disabled {
    border: 1px solid #DBDBDB;
    background-color: #F2F2F2;
    color: #DBDBDB;
    cursor: default;
  }

  &:focus {
    box-shadow: $focus-box-shadow
  }
}
