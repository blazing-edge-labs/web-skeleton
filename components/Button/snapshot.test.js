import React from 'react'
import renderer from 'react-test-renderer'
import Button from './'

describe('Button component snapshot', () => {
  it('renders basic required data', () => {
    const tree = renderer.create(<Button>Login</Button>)

    expect(tree.toJSON()).toMatchSnapshot()
  })

  it('renders empty button', () => {
    const tree = renderer.create(<Button empty>Login</Button>)

    expect(tree.toJSON()).toMatchSnapshot()
  })
})
