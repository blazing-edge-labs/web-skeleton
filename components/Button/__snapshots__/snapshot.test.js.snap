// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component snapshot renders basic required data 1`] = `
<button
  className="button "
>
  Login
</button>
`;

exports[`Button component snapshot renders empty button 1`] = `
<button
  className="button "
  empty={true}
>
  Login
</button>
`;
